### DISCUSSION TITLE
Documentation and/or Example for Nestjs?

### DISCUSSION THREAD

**Discussion by arimgibson** [hayes/pothos#1104] [Ideas]
I'm on the fence of whether this would provide value or not... thought I'd throw it out there as I'd be happy to PR if this is something that seems useful.

Pros:
- Nestjs is widely used
- Nest's own documentation on code-first GraphQL falls short, potential trap for beginners
  - Something I intend to take up with them as well
  - Easy to assume from their docs that you'd have to print the schema to a file before usage in Nest, which is false
  - No mention in their code-first docs about passing a schema (e.g. `builder.toSchema()`) directly to Nest, but it's easy when you notice the `schema` property which isn't mentioned in their docs
- More docs/examples, when organized correctly, rarely hurt
- Helpful for SEO purposes, could drive more users to Pothos who are intimated by having to figure out compatibility with Nest on their own

Cons
- Adds something else that needs to be maintained
- Could be cluttering with unnecessary docs/example for something not super hard once you get into it

I'm personally leaning towards having docs instead of an example, but I'm open to what others think is best. If the doc route makes sense, let me know where's the best place to put it. My suggestion would be in the "guide" section on the website, but I also don't want it to get lost there or seem out of place. :smiley:

**Comment by hayes** [comment]
Always happy to have more examples.

I think what I would probably do is start with a high quality example in the examples section of the repo, with any documentation added as a readme in the example directory.

Instead of documenting different frameworks directly on the website, it might be better to just create an index page with descriptions and links out to specific example projects

**Reply by arimgibson** [reply]
Got it, makes sense! I'll work on that example when I get a chance. Planning on using Mercurius and Fastify as the GraphQL backend, as they've got the best performance supported directly through NestJs integrations
